resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/config.scala

```diff
@@ -325,6 +325,17 @@ package object config extends Logging {
     .bytesConf(ByteUnit.MiB)
     .createWithDefaultString("512m")
 
+  /* Shared driver and executor configuration. */
+
+  private[spark] val MEMORY_OVERHEAD_FACTOR =
```

Comment:
Memory overhead is not specific to yarn, but applies to all resource managers.
Let us add this to core's `org.apache.spark.internal.package.scala`
Also, split it by driver and executor as well, to independently control them.